hugegraph-core/src/main/java/com/baidu/hugegraph/backend/query/ConditionQuery.java

```diff
@@ -522,4 +569,144 @@ private static Object convertNumberIfNeeded(Object value) {
         INDEX,
         INDEX_FILTER
     }
+
+    public static final class Element2IndexValueMap {
+
+        private final Map<Id, Set<LeftIndex>> leftIndexMap;
+        private final Map<Id, Map<Id, Set<Object>>> filed2IndexValues;
+        private Id selectedIndexField;
```

Comment:
seems not work as expected. JointQuery will do index query one by one before test element. So `selectedIndexField` will always be same.

Reply:
```
IdHolder holder = this.doIndexQuery(indexLabel, query);
            if (resultHolder == null) {
                resultHolder = holder;
                this.storeSelectedIndexField(indexLabel, query);
            }
            assert this.indexIntersectThresh > 0; // default value is 1000
            Set<Id> ids = ((BatchIdHolder) holder).peekNext(
                          this.indexIntersectThresh).ids();
            if (ids.size() >= this.indexIntersectThresh) {
                // Transform into filtering
                filtering = true;
                query.optimized(OptimizedType.INDEX_FILTER);
            } else if (filtering) {
                assert ids.size() < this.indexIntersectThresh;
                resultHolder = holder;
                this.storeSelectedIndexField(indexLabel, query);
                break;
            }
```

Case:
- the first index's ids size > indexIntersectThresh
- the second index's ids size < indexIntersectThresh